sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/cpp/CppGenerator.java

```diff
@@ -1703,10 +1722,18 @@ private CharSequence generateFileHeader(
             "#if __cplusplus >= 201703L\n" +
             "#  include <string_view>\n" +
             "#  define SBE_NODISCARD [[nodiscard]]\n" +
+            "#  define SBE_USE_STRING_VIEW 1\n" +
```

Comment:
Defines should be protected with `#ifndef XYZ` or `#if !define(XYZ)` like `__STDC_LIMIT_MACROS` below.